### ISSUE TITLE
Ethers.js Adapters Code Sample Doesn't Work

### ISSUE THREAD

**Issue by frankfka** [wevm/wagmi#3398]
### What is the type of issue?

Documentation is incorrect

### What is the issue?

I'm currently scoping an upgrade to Wagmi V2 and trying to use the provided adapter for Ethers V6.

Our setup is quite simple, with the suggested config of:
```
chains: [arbitrumSepolia],
transports: {
   [arbitrumSepolia.id]: http()
}
```
And calling something like: `useEthersProvider({ chainId: arbitrumSepolia.id })`, we get an error from JsonRpcProvider because it attempts to connect to `localhost:8545`.

Upon digging, it comes from: `return new JsonRpcProvider(transport.url, network)` in the documentation - essentially, `transport.url` is undefined (I can't find a relevant `url` entry in the TS autocomplete either).

Instead, something like `return new JsonRpcProvider(chain.rpcUrls.default.http[0], network)` works fine, but I don't know if there is a better method.

### Where did you find it?

https://wagmi.sh/react/guides/ethers

**Comment by jxom** [comment]
I can't seem to reproduce this in the [Wagmi Playground](https://github.com/wevm/wagmi/tree/main/playgrounds/vite-react) by copy-pasting the code from the guide. Mind providing a minimal reproducible example in https://new.wagmi.sh?

Will re-open once a minimal repro is provided!

**Comment by frankfka** [comment]
@jxom Please see the following: https://stackblitz.com/edit/new-wagmi-skbrda?file=src%2FviemToEthers.ts 

I'm logging `transport.url`, which shows as `undefined`. And if you inspect the console logs you'll see the following:
<img width="918" alt="image" src="https://github.com/wevm/wagmi/assets/31530056/b0afd885-ede2-41bb-a190-4dc4daea22bc">

**Comment by jxom** [comment]
Looks like Viem wasn't attaching the derived URL to the transport. Fixed here: https://github.com/wevm/viem/commit/991b29367ea2ef594d1daa1c3f82fb5f1dffc046

Should be released soon in Viem 2.0.4.

**Comment by frankfka** [comment]
> Looks like Viem wasn't attaching the derived URL to the transport. Fixed here: [wevm/viem@991b293](https://github.com/wevm/viem/commit/991b29367ea2ef594d1daa1c3f82fb5f1dffc046)
> 
> Should be released soon in Viem 2.0.4.

Thank you!
